### PR TITLE
Add drag pan type

### PR DESCRIPTION
**Pull request by ghsteff**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

The only way to pan the chart is with scrolling

## What is the new behavior?
You can pan charts by dragging the screen with `panType='drag'`. This will also enable zooming with the mouse wheel

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information

https://github.com/reaviz/reaflow/assets/40581813/9decd531-76ad-45e4-ab3a-618f79dfb395

### CODE REVIEW

**Review by amcdnl — APPROVED** [review]

File: src/Canvas.tsx [hunk 1]
```diff
@@ -244,11 +274,34 @@ const InternalCanvas: FC<CanvasProps & { ref?: Ref<CanvasRef> }> = forwardRef(({
     }
   }, [createDragNodeChildren, dragNodeData, layout?.children]);
 
+  useEffect(() => {
+    const zoomOnWheelScroll = (e: WheelEvent) => {
```
[the remark this answers is not in the document]
Comment (amcdnl) [reply src/Canvas.tsx:278]: I would also combine the gestures into 1 and use the wheel event they provide - https://use-gesture.netlify.app/docs/gestures/#handling-multiple-gestures-at-once

[the remark this answers is not in the document]
Comment (ghsteff) [reply src/Canvas.tsx]: ah nice that's cleaner

[the remark this answers is not in the document]
Comment (ghsteff) [reply src/Canvas.tsx]: the pinch zoom is already being applied here https://github.com/reaviz/reaflow/blob/85ec43cad3e179e9d586f74567932144fc44fb27/src/utils/useZoom.ts#L52-L67

File: src/Canvas.tsx [hunk 2]
```diff
@@ -313,7 +302,6 @@ const InternalCanvas: FC<CanvasProps & { ref?: Ref<CanvasRef> }> = forwardRef(({
       }}
       onMouseEnter={onMouseEnter}
       onMouseLeave={onMouseLeave}
-      {...bind()}
```
Comment (amcdnl) [inline src/Canvas.tsx]: You don't need this because u set the `domTarget` correct?

Comment (ghsteff) [reply src/Canvas.tsx]: Correct

**Pull request merged**